hs/src/Reach/Eval/Core.hs

```diff
@@ -2413,9 +2413,11 @@ evalApplyVals rator randvs =
     SLV_CloTyped clo_at sc tf -> do
       at <- withAt id
       let isDLVar = \case { SLV_DLVar _ -> True; _ -> False } . snd
-      unless (all isDLVar randvs) $
-        void $ assertRefinedArgs randvs at tf
-      evalApplyClosureVals clo_at sc randvs
+      let ct = if all isDLVar randvs then CT_Assume True else CT_Assert
```

Comment:
I don't get why this isn't always `CT_Assume True`?

Reply:
Typed closures can be applied within a Reach program, so it should assert the static arguments satisfy the pre condition instead of assuming

Reply:
This is not a good check for that, because I could apply the function in Reach and still send it all DLVars... like the result of a publish. So this should really be based on something else...

Reply:
Easier test would be a `forall`, which returns a `DLVar`

Reply:
I don't understand using `forall` in this scenario?

Reply:
`forall` returns a `DLVar`, although... I can't think of a way to actually use it usefully. Sorry

Reply:
I ended up using the mode to determine whether to use assume or assert. In `SLM_Module`, the only dynamic values that'll exist are arguments from the frontend to exported functions